server/http/controller.go

```diff
@@ -29,19 +30,20 @@ import (
 var IDGen = id.NewRandGenerator()
 
 type controller struct {
-	tracer          trace.Tracer
-	testDB          model.Repository
-	runner          runner
-	newTraceDBFn    func(ds datastoreresource.DataStore) (tracedb.TraceDB, error)
-	mappers         mappings.Mappings
-	triggerRegistry *trigger.Registry
-	version         string
+	tracer                trace.Tracer
+	testDB                model.Repository
+	runner                runner
+	newTraceDBFn          func(ds datastoreresource.DataStore) (tracedb.TraceDB, error)
+	environmentRepository environment.Repository
```

Comment:
you are only using the `Get` function, so maybe instead of depending on the repo directly, you could depend on a local interface:
```
type envGetter interface {
  Get(context.Context, id.ID) (environment.Environment, error) 
}

type controller struct {
  //...
  environmentRepository envGetter
}
```

that way if we need to pass a mock or any other substitution, we only need to implement that func and not the entire repo